app/script/components/participantAvatar.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.ParticipantAvatar = class ParticipantAvatar {
+  constructor(params, component_info) {
+    this.participant = ko.unwrap(params.participant);
+    this.isUser = this.participant instanceof z.entity.User && !this.participant.isBot;
+    this.isService = this.participant instanceof z.integration.ServiceEntity || this.participant.isBot;
+
+    this.delay = params.delay;
+    this.size = params.size;
+    this.element = $(component_info.element);
+
+    if (this.size) {
+      this.element.addClass(`${this.isService ? 'service' : 'user'}-avatar-${this.size}`);
```

Comment:
how many things depend on this CSS class? Can we move from individual service and user classes to particpant classes for the sizing or do other things depend on it?

Reply:
the less files will definitely be merged later on.